### PR TITLE
ENH: Timestamp.normalize support non-nano

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by mroeschke — COMMENTED** [review]
Also minor, I noticed some of the typings in the cython code for timezones are e.g. `tzinfo tz` when it looks like it should be `tzinfo | None tz` like in `_from_value_and_reso`.

File: pandas/_libs/tslibs/timestamps.pyx [hunk 1]
```diff
@@ -856,12 +856,11 @@ cdef class _Timestamp(ABCTimestamp):
...
             int64_t normalized
             int64_t ppd = periods_per_day(self._reso)
-
-        if self._reso != NPY_FR_ns:
-            raise NotImplementedError(self._reso)
+            _Timestamp ts
 
         normalized = normalize_i8_stamp(local_val, ppd)
-        return Timestamp(normalized).tz_localize(self.tzinfo)
+        ts = type(self)._from_value_and_reso(normalized, reso=self._reso, tz=None)
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timestamps.pyx:862]: no

File: pandas/_libs/tslibs/timestamps.pyx [hunk 2]
```diff
@@ -2035,6 +2034,8 @@ default 'raise'
         NaT
         """
         if self._reso != NPY_FR_ns:
+            if tz is None and self.tz is None:
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timestamps.pyx:2037]: yes

**Comment by jbrockmendel** [comment]
> Also minor, I noticed some of the typings in the cython code for timezones are e.g. tzinfo tz when it looks like it should be tzinfo | None tz like in _from_value_and_reso.

good catch, will edit in an upcoming PR

**Review by mroeschke — APPROVED** [review]

**Pull request merged**